art/attacks/svm_attack.py

```diff
@@ -0,0 +1,264 @@
+        if self.max_iters <= 1:
+            raise ValueError("Value of max_iters must be strictly positive")
+
+
+def generate_attack_point(classifier, x_train, y_train, x_val, y_val, x_attack, y_attack, step, eps, max_iters):
+    """
+    Generate a single poison attack the model, using x_val and y_val as validation points.
+    The attack begins at the point init_attack. The attack class will be the opposite of the model's
+    classification for init_attack.
+
+    :param classifier: a trained SVC classifier
+    :type classifier: `art.classifiers.ScikitlearnSVC`
+    :param x_train: training data
+    :type x_train: `np.ndarray`
+    :param y_train: training labels
+    :type y_train: `np.ndarray`
+    :param x_val: validation data
+    :type x_val: `np.ndarray`
+    :param y_val: validation labels
+    :type y_val: `np.ndarray`
+    :param x_attack: the initial attack point
+    :type x_attack: `np.ndarray`
+    :param y_attack: the initial attack label
+    :type y_attack: `np.ndarray`
+    :param step: the step size
+    :type step: `float`
+    :param eps: the min difference in loss to aim for
+    :type eps: float
+    :param max_iters: the maximum number of iterations the attack should run
+    :type max_iters: int
+    :return: a tuple containing the final attack point and the poisoned model
+    :rtype: (`np.ndarray`, `art.classifiers.ScikitlearnSVC`)
+    """
+    poisoned_model = SVC(kernel=classifier._model.kernel, probability=classifier._model.probability)
+    y_t = np.argmax(y_train, axis=1)
+    poisoned_model.fit(x_train, y_t)
+    y_a = np.argmax(y_attack)
+    attack_point = np.expand_dims(x_attack, axis=0)
+    g = poisoned_model.decision_function(x_val)
+    k_values = np.where(-g > 0)
+    new_p = np.sum(g[k_values])
+    old_p = np.copy(new_p)
+    i = 0
+
+    while new_p - old_p < eps and i < max_iters:
+        old_p = new_p
+        poisoned_input = np.vstack([x_train, attack_point])
+        poisoned_labels = np.append(y_t, y_a)
+        poisoned_model.fit(poisoned_input, poisoned_labels)
+
+        unit_grad = normalize(attack_gradient(poisoned_model, attack_point, x_val[k_values], y_val[k_values]))
+        attack_point += step * unit_grad
+        lower, upper = classifier.clip_values
+        new_attack = np.clip(attack_point, lower, upper)
+        new_g = poisoned_model.decision_function(x_val)
+        k_values = np.where(-new_g > 0)
+        new_p = np.sum(new_g[k_values])
+        i += 1
+        attack_point = new_attack
+
+    poisoned_input = np.vstack([x_train, attack_point])
+    poisoned_labels = np.append(y_t, y_a)
+    poisoned_model.fit(poisoned_input, poisoned_labels)
+    poisoned_model = ScikitlearnSVC(model=poisoned_model, clip_values=classifier.clip_values,
+                                    defences=classifier.defences, preprocessing=classifier.preprocessing)
+    return attack_point, poisoned_model
+
+
+def predict_sign(model, input):
```

Comment:
Could you make this an instance method belonging to the class?
If yes, remove the redundant arguments and use `self`.
It looks like this function is called only from within this class. If you don't intend users to call this function, you could add a underscore `_` at the beginning of the method name.

Reply:
`generate_attack_point` (a static function) also depends on this, so it too must be static